[project-base] - add hover timeout to horizontal menu (#1564)
shopsys/shopsys: upgrade/UPGRADE-v9.0.0-dev.md

             - public function __construct(string $entityName, int $entityId, string $type, string $temporaryFilename, int $position)
             + public function __construct(string $entityName, int $entityId, string $type, string $temporaryFilename, string $uploadedFilename, int $position)
            ```

 - There is a new base html layout with horizontal menu and product filter placed in left panel, for detail information see [the separate article](upgrade-instructions-for-base-layout.md)

> 

Reviewer: ```suggestion
```

- add hover timeout to horizontal menu ([#1564](https://github.com/shopsys/shopsys/pull/1564))
    - you can skip this task if you have your custom design
    - move loader to hidden submenu - so it can not interrupt hover `src/Resources/scripts/frontend/categoryPanel.js`
        ```diff
          function loadCategoryItemContent ($categoryItem, url) {
              Shopsys.ajax({
        -          loaderElement: $categoryItem,
        +          loaderElement: $categoryItem.find('.js-category-list-placeholder'),
        ```

> 

Reviewer: ```suggestion
```

    - add new js plugin hoverIntent v1.10.1 `src/Resources/scripts/frontend/plugins/jquery.hoverIntent.js` (https://github.com/shopsys/shopsys/tree/master/project-base/src/Resources/scripts/frontend/plugins/jquery.hoverIntent.js)

> 

Reviewer: ```suggestion
```

    - add new js component `src/Resources/scripts/frontend/components/hoverIntent.js` (https://github.com/shopsys/shopsys/tree/master/project-base/src/Resources/scripts/frontend/components/hoverIntent.js)

> 

Reviewer: ```suggestion
```

>     - add classes to hover menu `templates/Front/Content/Category/panel.html.twig`

Reviewer: ```suggestion
    - add class and data attributes to hover menu `templates/Front/Content/Category/panel.html.twig`
```

Author: fixed, thx
